rating-display.js

```diff
@@ -0,0 +1,299 @@
+        console.log("Something went wrong");
+        this.rating = 0;
+      }
+    }
+  }
+  // as soon as the web component is loaded into the browser window
+  // the connectedCallback() method is called
+  connectedCallback() {
+    super.connectedCallback();
+    console.log("Connected Callback");
+    this.rating = 0;
+    this.get_rating(this.experimentURL, this.experimentName);
+  }
+  // get and set methods for the properties
+  get sheetName() {
+    return this._sheetName;
+  }
+  set sheetName(name) {
+    this._sheetName = name;
+    this.requestUpdate();
+  }
+  set spreadsheetID(id) {
+    this._spreadsheetID = id;
+    this.requestUpdate();
+  }
+  get spreadsheetID() {
+    return this._spreadsheetID;
+  }
+  set columnName(name) {
+    this._columnName = name;
+    this.requestUpdate();
+  }
+  get columnName() {
+    return this._columnName;
+  }
+  set columnValue(value) {
+    this._columnValue = value;
+    this.requestUpdate();
+  }
+  get columnValue() {
+    return this._columnValue;
+  }
+  get fullStars() {
+    return this._fullStars;
+  }
+  set fullStars(newVal) {
+    this._fullStars = newVal;
+    this.requestUpdate();
+  }
+  get halfStars() {
+    return this._halfStars;
+  }
+  set halfStars(newVal) {
+    this._halfStars = newVal;
+    this.requestUpdate();
+  }
+  set rating(newRating) {
+    console.log("New Rating is ", newRating);
+    this._rating = newRating;
+    this._roundRating = Math.round(2 * newRating) / 2;
+    if (this._roundRating % 1 === 0) {
+      this._fullStars = this._roundRating;
+      this._halfStars = 0;
+    } else {
+      this._fullStars = Math.floor(this._roundRating);
+      this._halfStars = 1;
+    }
+    console.log(this._fullStars, this._halfStars);
+    this.requestUpdate();
+  }
+  get rating() {
+    return this._rating;
+  }
+  set title(newTitle) {
+    this._title = newTitle;
+  }
+  get title() {
+    return this._title;
+  }
+  get numberOfStars() {
+    return this._numberOfStars;
+  }
+  set numberOfStars(newVal) {
+    this._numberOfStars = newVal;
+    this.requestUpdate();
+  }
+  //  constructor
+  constructor() {
+    super();
+    this._numberOfStars = 5;
+    if (this._roundRating % 1 === 0) {
+      this._fullStars = this._roundRating;
+      this._halfStars = 0;
+    } else {
+      this._fullStars = Math.floor(this._roundRating);
+      this._halfStars = 1;
+    }
+    const fa = document.createElement("link");
+    fa.rel = "stylesheet";
+    fa.type = "text/javascript";
+    fa.href = "https://unpkg.com/fontawesome@5.6.3/index.js";
+    document.head.appendChild(fa);
+  }
+  render() {
+    console.log(this._fullStars, this._halfStars);
+    const stars = [];
+    for (let i = 0; i < this._fullStars; i++) {
+      stars.push(
+        html`<img src="./images/star.svg" class="star-images"></img>`
+        // html`<img src="http://localhost:5500/images/star.svg" class="star-images"></img>`
+      );
+    }
+    for (let i = 0; i < this._halfStars; i++) {
```

Comment:
Why do we need a for loop here? There can only be 1 half star at the maximum

Reply:
yep, could be replaced by a conditional, fixed and updated.